src/intersection-observer-polyfill.js

```diff
@@ -233,60 +259,66 @@ export class IntersectionObserverPolyfill {
           moveLayoutRect(opt_iframe, -opt_iframe.left, -opt_iframe.top);
     }
 
-    // Normalize container LayoutRect to be relative to page
-    let elementRect;
-    let ownerRect = null;
-
-    // If opt_iframe is provided, all LayoutRect has position relative to
-    // the iframe.
-    // If opt_iframe is not provided, all LayoutRect has position relative to
-    // the host document.
-    elementRect = this.element_.getLayoutBox();
-    const owner = this.element_.getOwner();
-    ownerRect = owner && owner.getLayoutBox();
+    this.observeEntries_.forEach(entry => {
```

Comment:
`#forEach` is considerably slower than a `for` loop due to edge cases (sparse array crap, mainly). https://jsperf.com/for-vs-foreach/2